Ray.jl/src/runtime.jl

```diff
@@ -115,7 +115,10 @@ function submit_task(f::Function, args...)
     return GC.@preserve args rayjll._submit_task(project_dir(), fd, object_ids)
 end
 
-function task_executor(ray_function, ray_objects)
+function task_executor(ray_function, returns, ray_objects)
+    returns = rayjll.cast_to_buffer(returns)
+    ray_objects = rayjll.cast_to_rayobject(ray_objects)
```

Comment:
These function names are rather inaccurately named. Would it be better to do:
```suggestion
    returns = rayjll.cast_to_returns(returns)
    ray_objects = rayjll.cast_to_arg_objects(ray_objects)
```
Maybe we should rename `ray_objects` to `arg_objects`? The name `ray_objects` doesn't indicate args at all.

Reply:
yeah I'm fine with `arg_objects`. But I reworded `cast_to_returns` to `cast_to_buffer` because it's only incidental to the fact that it's used for "task returns" in the end.

Reply:
I'm kinda on the fence here; on the one hand, the variable names here could be used to communicate what the role these values play (I'd change `ray_objects` to `task_args` FWIW), and the function names then are free to describe what kind of values are being generated.  on the other hand, if we expect to _only_ use these for returns/args, then we shoudl just call them that.

Reply:
Ok I take both your points and will rename the functions. will also rename `ray_object` to `task_args` while I'm at it.